### ISSUE TITLE
tvrplus.ro : Public TV of Romania

### ISSUE THREAD

**Issue by karlo2105** [streamlink/streamlink#586]
### Checklist

- [ ] This is a bug report.
- [x] This is a plugin request.
- [ ] This is a feature request.
- [ ] I used the search function to find already opened/closed issues or pull requests.

### Description

TVR is Romanian public service consisting of 11 channels streamed [online](http://www.tvrplus.ro/). Channels are geoblocked locally except TVR International. 

- TVR 1 : `http://www.tvrplus.ro/live-tvr-1`
- TVR HD : `http://www.tvrplus.ro/live-tvr-hd`
- TVR 2 : `http://www.tvrplus.ro/live-tvr-2`
- TVR 3 : `http://www.tvrplus.ro/live-tvr-3`
- TVR Int. : `http://www.tvrplus.ro/live-tvr-international`
- TVR Moldova : `http://www.tvrplus.ro/live-tvr-moldova`
- TVR Cluj : `http://www.tvrplus.ro/live-tvr-cluj`
- TVR Craivova : `http://www.tvrplus.ro/live-tvr-craiova`
- TVR Iasi : `http://www.tvrplus.ro/live-tvr-iasi`
- TVR Targu Mures : `http://www.tvrplus.ro/live-tvr-targu-mures`
- TVR Timisoara : `http://www.tvrplus.ro/live-tvr-timisoara`

### Expected / Actual behavior

They propose two protocols : rtmp and HLS. Streaming addresses can be found in page source code. 

```
sources: [
file: "rtmp://fms84.mediadirect.ro/live3?publisher=4&token=1a6d61fa3f6ca91751b27989237a53d6e618c8462b241bf1mp4:tvr"
			{
file: "http://fms84.mediadirect.ro:1937/live3/_definst_/tvr/playlist.m3u8?publisher=4&token=1a6d61fa3f6ca91751b27989237a53d6e618c8462b241bf1",
			}]
```
However I didn't manage to launch rtmp stream just HLS. If rtmp is not active, it can be dropped. 

### Reproduction steps / Stream URLs to test

```
streamlink -p "%homedrive%\Users\Ddr\Downloads\VLCPortable\App\vlc\vlc" "hlsvariant://http://fms74.mediadirect.ro:1937/live3/_definst_/tvr/playlist.m3u8?publisher=4&token=16276249ae8191ffed3f87094929eae4de0c22fa0bafcf70" best
[cli][info] Found matching plugin stream for URL hlsvariant://http://fms74.media
direct.ro:1937/live3/_definst_/tvr/playlist.m3u8?publisher=4&token=16276249ae819
1ffed3f87094929eae4de0c22fa0bafcf70
[cli][info] Available streams: 91k (worst), 576p (best)
[cli][info] Opening stream: 576p (hls)
[cli][info] Starting player: C:\Users\Ddr\Downloads\VLCPortable\App\vlc\vlc
[cli][info] Player closed
[cli][info] Stream ended
[cli][info] Closing currently open stream...

```
### Environment details (operating system, python version, etc.)

It was tested on streamlink portable on Win 7 Pro.

PS : I don't know how to put label in key words.

**Comment by beardypig** [comment]
Simple. Thanks for the detailed request, makes it much easier :)

**Comment by beardypig** [comment]
I could only test the international stream, perhaps you can test out the others?

**Comment by karlo2105** [comment]
I just did it for every geo-locked channel it displays such message : 

```
streamlink -l debug -p "%homedrive%\Users\Ddr\Downloads\VLCPortable\App\vlc\vlc" --https-proxy "81.181.81.26:3128" "http://www.tvrplus.ro/live-tvr-1" best
[cli][info] Found matching plugin tvrplus for URL http://www.tvrplus.ro/live-tvr
-1
error: No playable streams found on this URL: http://www.tvrplus.ro/live-tvr-1
```

**Comment by karlo2105** [comment]
I confirm it's working fine. SD channels air in 576p. It's funny their HD channel is only 480p. 

```
streamlink --http-proxy "94.177.253.224:8080" "http://www.tvrplus.ro/live-tvr-hd" best
[cli][info] Found matching plugin tvrplus for URL http://www.tvrplus.ro/live-tvr
-hd
[cli][info] Available streams: 95k (worst), 480p (best)
[cli][info] Opening stream: 480p (hls)
[cli][info] Starting player: "C:\Program Files\VideoLAN\VLC\vlc.exe"
[cli][info] Player closed
[cli][info] Stream ended
[cli][info] Closing currently open stream...
```

**Comment by gravyboat** [comment]
https://github.com/streamlink/streamlink/pull/587 is merged so I'm going to close this. Thanks for testing it out @karlo2105.

**Comment by daeongithub** [comment]
Okay so I was able to make an account on TVR+ to start watching the channels online and through my Livestream Downloader thing I was able to get a link, however the link has a session ID. But this is a start in the right direction for getting the TVR channels back, no? Try this link and see if it works, or see if anyone else is successful!

It seems they changed their livestream provider to Dejacast. https://milos.cdn.dejacast.com/live//tvr1/tvr1.m3u8?from=1767299281&to=1767320881&session=vUs5GjI5oi2eQ6Px&source=cms&token=7vlmfgQxoxXm6qI24rmz4QrWSs4=

However, if I take off all the rest of the strings then it doesn't load, and I tried putting the link into my copy of the IPTV list and confirmed it doesn't load on TV, just on my Google Chrome Web Browser. Maybe someone else can go back into this and see if it works?
